src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -120,26 +121,29 @@ template Instruction
         approval
       allocate Instruction.Allocate{actors; allocation} = do
         let mustAuthorize = mustAuthorizeHelper actors
-        assertMsg ("allocation must be new" <> messageSuffix) $ allocation /= this.allocation
-        -- undisclose previous allocation
-        obCid <- undisclosePreviousAllocation
+        assertMsg ("allocation must be new " <> context) $ allocation /= this.allocation
+        -- release previous allocation
+        obCid <- releasePreviousAllocation
         -- allocate
         allocationDisclosed <- case allocation of
           Pledge baseCid -> do
             base <- fetch baseCid
             let senderAccount = getAccount base
             vSenderAccount <- view <$> fetchInterfaceByKey @Account.R @Account.I senderAccount
-            discloseA senderAccount True
-            baseCid <- discloseB vSenderAccount.controllers.outgoing settlers baseCid
+            discloseAccount senderAccount True
+            baseCid <- disclosePledge settlers vSenderAccount.controllers.outgoing baseCid
             mustAuthorize vSenderAccount.controllers.outgoing
-            mustAuthorize $ Base.getLockers base
             mustBe Custodian senderAccount.custodian
             mustBe Sender senderAccount.owner
             getAmount base === routedStep.quantity.amount
             getInstrument base === routedStep.quantity.unit
-            pure $ Pledge baseCid
+            lockedCid <- exercise baseCid Base.Acquire with
+              newLockers = requestors <> vSenderAccount.controllers.outgoing
```

Comment:
I wonder if we should consider this a breaking change to the `Instruction` as it changes previous functionality

Reply:
right, I bumped the major instead